agent/job.py

```diff
@@ -195,12 +195,14 @@ def wrapper(wrapped, instance: Base, args, kwargs):
                 instance.job_record.success(result)
             return result
         agent_job_id = get_agent_job_id()
+        agent_job_timeout = kwargs.pop("agent_job_timeout", None)
         instance.job_record.enqueue(name, wrapped, args, kwargs, agent_job_id)
+        final_timeout = agent_job_timeout or timeout or instance.config["job_timeout"] or DEFAULT_TIMEOUT
```

Comment:
```
 instance.config["job_timeout"]
```

In case of `Bench` class, the config_file is linked to config of bench.
That class has a `server` variable which you can use though to access the agent config.

Also, use `.get("job_timeout")` instead of `[...]` , as this can raise error if the variable isn't there in config,